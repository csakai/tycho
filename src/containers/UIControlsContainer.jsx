import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as Actions from '../actions/UIControlsActions';
import ReduxService from '../services/ReduxService';
import UIControls from '../components/UIControls';

export class UIControlsContainer extends React.Component {

    static propTypes = {
        time: PropTypes.number
    }

    /**
     * Toggles the settings pane.
     */
    toggleSettings = () => {
        this.props.action.toggleSettings(!this.props.settingsActive);
    }

    /**
     * Opens the modal of the given type and hides the controls.
     *
     * @param {String} type - modal type to open
     */
    openModal = (type) => {
        this.props.action.toggleModal(type);
        this.props.action.setUIControls(false);
    }

    render() {
        return (
            <UIControls
                openModal={this.openModal}
                toggleSetting={this.toggleSettings}
                {...this.props}
                {...this.props.action}
            />
        );
    }
}

export default connect(
    ReduxService.mapStateToProps(
        'uiControls.speed',
        'uiControls.zoom',
        'uiControls.scale',
        'uiControls.controlsEnabled',
        'uiControls.settingsActive',
        'label.targetName',
        'data.pageText',
        'animation.time'
    ),
    ReduxService.mapDispatchToProps(Actions)
)(UIControlsContainer);
