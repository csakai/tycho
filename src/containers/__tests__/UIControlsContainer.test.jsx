import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {UIControlsContainer} from '../UIControlsContainer';

describe('UIControls Container', () => {
  let component, container;

  beforeEach(() => {
    component = shallow(<UIControlsContainer
      action={{
        changeSpeed: jest.fn()
      }}
    />);
    container = component.instance();
  });

  describe('toggleSettings()', () => {
    beforeEach(() => {
      container.props = {
        action: {
          toggleSettings: jest.fn()
        },
        settingsActive: false
      };
    });

    it('should open the settings pane', () => {
      const spy = jest.spyOn(container.props.action, 'toggleSettings');

      container.toggleSettings();

      expect(spy).toHaveBeenCalled();
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith(!container.props.settingsActive);
    });
  });

  describe('openModal()', () => {
    beforeEach(() => {
      container.props = {
        action: {
          toggleModal: jest.fn(),
          setUIControls: jest.fn()
        }
      };
    });

    it('should open the modal', () => {
      const type = 'TEST_MODAL';
      const spy = jest.spyOn(container.props.action, 'toggleModal');

      container.openModal(type);

      expect(spy).toHaveBeenCalled();
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith(type);
    });

    it('should hide the UI Controls', () => {
      const spy = jest.spyOn(container.props.action, 'setUIControls');

      container.openModal();

      expect(spy).toHaveBeenCalled();
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith(false);
    });
  });

  describe('render()', () => {
    it('should successfully render the ui controls container', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
