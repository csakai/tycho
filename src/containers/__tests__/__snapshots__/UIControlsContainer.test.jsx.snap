// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIControls Container render() should successfully render the ui controls container 1`] = `
<div>
  <div
    className="uicontrol uicontrol--scales"
  >
    <ScaleSlider
      label="Time speed: × 10^"
      max={10}
      min={0}
      onChange={[Function]}
    />
    <ScaleSlider
      label="Planet scale: × "
      max={10}
      min={1}
    />
  </div>
  <div
    className="uicontrol uicontrol--datetime"
  >
    <DatePicker />
  </div>
  <div
    className="uicontrol uicontrol--left-bar"
  >
    <div
      className="uicontrol__button"
    />
    <ZoomSlider />
    <div
      className="uicontrol__button"
    />
  </div>
</div>
`;
